pkg/goDB/DBWorkManager.go

```diff
@@ -219,201 +205,197 @@ func (w *DBWorkManager) ExecuteWorkerReadJobs(ctx context.Context, mapChan chan
 
 // Block evaluation and aggregation -----------------------------------------------------
 // this is where the actual reading and aggregation magic happens
-func (w *DBWorkManager) readBlocksAndEvaluate(ctx context.Context, workload DBWorkload, resultMap hashmap.AggFlowMapWithMetadata) error {
+func (w *DBWorkManager) readBlocksAndEvaluate(workload DBWorkload, resultMap hashmap.AggFlowMapWithMetadata) error {
```

Comment:
The implication being that single workloads cannot be cancelled anymore. E.g.: one full day has to complete processing before exiting, correct?

Was this change related to performance optimization?

Reply:
Indeed, the point was that the current code basically checks for the cancellation just _before_ "entering" a day block and then again on each block. The performance implication is not relevant, it was more a matter of removal of some practically unneeded check. I mean, what's the scenario? If the query is short you won't have time to even hit Ctrl^C, and if it's long then it's totally fine if you cancel after a workload is completed. But I don't have super strong feelings about it TBH...